cmd/api-linter/summary_cli.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/googleapis/api-linter/lint"
+)
+
+func createSummary(responses []lint.Response) (summary LintSummary) {
+	summary.numSourceFiles = len(responses)
+	summary.violationData = make(map[string]map[string]bool)
+	for _, response := range responses {
+		pathToAdd := string(response.FilePath)
+		problems := response.Problems
+		for _, currentProb := range problems {
+			ruleName := string(currentProb.RuleID)
+			if existingPaths, ok := summary.violationData[ruleName]; ok {
+				if _, isExist := existingPaths[pathToAdd]; !isExist {
+					existingPaths[pathToAdd] = true
+				}
+			} else {
+				summary.longestRuleLen = max(summary.longestRuleLen, len(ruleName))
+				summary.violationData[ruleName] = map[string]bool{pathToAdd: true}
+			}
+		}
+	}
+	return summary
+}
+
+// Given a pointer to a summary of lint responses, and output location,
+// this functions will return a []byte which shows the percentage of files failing for each rule
+func emitSummary(summary *LintSummary) ([]byte, error) {
+	var sb strings.Builder
```

Comment:
Nice to have: use text/template for all this.

Reply:
Done